app/gui/view/graph-editor/src/component/visualization/foreign/java_script/visualization.js

```diff
@@ -55,6 +63,18 @@ export class Visualization {
         return this.__preprocessorModule__
     }
 
+    /**
+     * Set the arguments of the preprocessor function.
+     *
+     * @param arguments the arguments passed to the preprocessor function.
+     */
+    setArguments(...args) {
+        if (args !== this.__preprocessorArguments__) {
+            this.__preprocessorArguments__ = args
+            this.__emitPreprocessorChange__()
```

Comment:
Ideally changing only the `setArguments` shall be a _lightweight_ operation. Something that invokes already existing code on the engine side, without re-parsing the code. But I guess this is provided _by definition_ when using method pointers...